Fix (some) components

codyphobe opened the pull request:
- Fix doors
- Fix Sabotage timers
- Fix MeetingHud (see also #8 for chat notes)

codyphobe committed:
Fix doors and sabotage timers

SomeoneWeird commented on lib/packets/PacketElements/Component.ts:
```diff
@@ -130,12 +130,14 @@ SYSTEM_HANDLER.set(SystemType.Communications, {
 		if("IsSabotaged" in curr && "IsSabotaged" in old) {
 			// assume SimpleComsSystem
-			return curr.IsSabotaged == curr.IsSabotaged
+			return curr.IsSabotaged != curr.IsSabotaged
```
Is one of these supposed to be `old`?

codyphobe replied:
Yep, missed that one when copying into GitHub

codyphobe committed:
Compare old with current for comms sabotage

codyphobe committed:
Fix MeetingHud

Because the `MeetingHud` tells us which bits are dirty when parsing, and because we set the player's `StateByte` only if it's dirty, we can deduce which bits are dirty when writing solely based on whether or not the player *has* a `StateByte` set.

codyphobe committed:
Actually fix MeetingHud

It would help if I copied all of the changes over. Perhaps I should just update my editor to use tabs instead of spaces temporarily so I don't have to copy over to the GitHub editor...
